document-store/src/integrationTest/java/org/hypertrace/core/documentstore/utils/Utils.java

```diff
@@ -92,4 +96,20 @@ public static Map<String, Object> convertDocumentToMap(Document document)
       throws JsonProcessingException {
     return OBJECT_MAPPER.readValue(document.toJson(), new TypeReference<>() {});
   }
+
+  public static void assertDocsAndSizeEqual(
+      Iterator<Document> documents, String filePath, int expectedSize) throws IOException {
+    String fileContent = readFileFromResource(filePath).orElseThrow();
+    List<Map<String, Object>> expected = convertJsonToMap(fileContent);
+
+    int actualSize = 0;
+    List<Map<String, Object>> actual = new ArrayList<>();
+    while (documents.hasNext()) {
+      actual.add(convertDocumentToMap(documents.next()));
+      actualSize++;
+    }
+
+    assertEquals(expected, actual);
+    assertEquals(expectedSize, actualSize);
```

Comment:
For every `aggregate()` result, can we also validate that the size matches the `count()` API? Ref.: `org.hypertrace.core.documentstore.mongo.MongoQueryExecutorIntegrationTest#assertSizeEqual(Query, java.lang.String)`

Reply:
I am counting the number of documents as part of the API - assertDocsAndSizeEqual

Reply:
Right! But, it is not hitting the `Collection.count()` method.
Therefore, the functionality of `count()` is not tested along with `aggregate()`.

Reply:
I have kept a Collection.count as separate API impl - https://github.com/hypertrace/document-store/issues/82
While doing that, we will add the tests.